core/src/sql.rs

```diff
@@ -0,0 +1,47 @@
+use std::{path::Path, str::FromStr};
+
+use sqlx::sqlite::{SqliteConnectOptions, SqliteJournalMode, SqlitePoolOptions};
+
+#[derive(Clone, Debug)]
+/// The sqlite pool is split into a writer and a reader pool.
+/// Wrapper around the sqlx::SqlitePool
+pub struct SqlitePool {
+    writer: sqlx::SqlitePool,
+    reader: sqlx::SqlitePool,
+}
+
+impl SqlitePool {
+    /// Connect to the sqlite database at the given path. Creates the database if it does not exist.
+    /// Uses WAL journal mode.
+    pub async fn connect(path: impl AsRef<Path>) -> anyhow::Result<Self> {
+        let db_path = format!("sqlite:{}", path.as_ref().display());
+        let conn_opts = SqliteConnectOptions::from_str(&db_path)?
+            .journal_mode(SqliteJournalMode::Wal)
+            .create_if_missing(true)
+            // .synchronous(sqlx::sqlite::SqliteSynchronous::Normal) // normally enough in WAL mode?
```

Comment:
Opinions on this setting? I also assume optimize on close is okay for both the RO and W pools.